File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution
{
public:
    bool isValid(string string)
    {
        stack<char> stack;

        if (string.size() == 0)
        {
            return true;
        }

        for (auto &&i : string)
        {
            if (i == '(' || i == '[' || i == '{')
            {
                stack.push(i);
            }
            else if ((i == ')' && !stack.empty() && stack.top() == '(') ||
                     (i == '}' && !stack.empty() && stack.top() == '{') ||
                     (i == ']' && !stack.empty() && stack.top() == '['))
            {
                stack.pop();
            }
            else
            {
                return false;
            }
        }
        if (stack.empty())
        {
            return true;
        }
        return false;
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function (s) {
  const stack = [];
  for (const i of s) {
    if (i == "(" || i == "[" || i == "{") {
      stack.push(i);
    } else if (
      (i === ")" && stack.length > 0 && stack[stack.length - 1] === "(") ||
      (i === "}" && stack.length > 0 && stack[stack.length - 1] === "{") ||
      (i === "]" && stack.length > 0 && stack[stack.length - 1] === "[")
    ) {
      stack.pop();
    } else {
      return false;
    }
  }

  if (stack.length === 0) {
    return true;
  }
  return false;
};